frontend/packages/ux-editor/src/components/config/componentSpecificContent/Summary2/Override/Summary2OverrideDisplaytype.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { useEffect } from 'react';
+import { StudioCard, StudioNativeSelect } from '@studio/components';
+import type {
+  Summary2OverrideConfig,
+  SummaryCustomTargetType,
+} from 'app-shared/types/ComponentSpecificConfig';
+import { useCustomConfigType } from './hook/useCustomConfigType';
+import { useTranslation } from 'react-i18next';
+
+export interface Summary2OverrideDisplaytypeProps {
+  override: Summary2OverrideConfig;
+  onChange: (override: Summary2OverrideConfig) => void;
+}
+
+export const Summary2OverrideDisplaytype = ({
+  override,
+  onChange,
+}: Summary2OverrideDisplaytypeProps) => {
+  const customConfigType = useCustomConfigType();
+  const { t } = useTranslation();
+
+  useEffect(() => {
+    if (!override.displayType) {
+      onChange({ ...override, displayType: 'string' });
+    }
+  }, [override, onChange]);
+
+  const handleCustomTypeChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const newType = event.target.value as SummaryCustomTargetType;
+    const updatedCustomType = { displayType: newType, componentId: override.componentId };
+    onChange(updatedCustomType);
+  };
+
+  return (
+    <StudioCard.Content>
+      <StudioNativeSelect
+        size='sm'
+        label={t('ux_editor.component_properties.overrides_type')}
+        onChange={handleCustomTypeChange}
+        value={override.displayType || 'string'}
+      >
+        {customConfigType.map((type) => (
+          <option key={type.value} value={type.value}>
+            {type.label}
+          </option>
+        ))}
+      </StudioNativeSelect>
+    </StudioCard.Content>
+  );
+};
```

Comment:
Maybe we can refactor this by splitting up the code more to increase readability? 😄 
Changed also include suggested changes above. 

```suggestion
  return (
    <StudioCard.Content>
      <StudioNativeSelect
        size='sm'
        label={t('ux_editor.component_properties.overrides_type')}
        onChange={handleCustomTypeChange}
        value={override.displayType || 'string'}
      >
        <CustomConfigTypeOptions />
      </StudioNativeSelect>
    </StudioCard.Content>
  );
};

const CustomConfigTypeOptions = (): ReactElement[] => {
   const customConfigTypes: CustomConfigType[] = useCustomConfigType();
   
   return customConfigTypes.map((type: CustomConfigType) => (
      <option key={type.value} value={type.value}>
         {type.label}
      </option>
   ))
}
```

Reply:
Fixed